dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCResponseHeadersTest.java

```diff
@@ -0,0 +1,1490 @@
+
+    @Before
+    public void setUpBeforeEachTest() throws Exception {
+    
+        Exception failure = null;
+        String consoleLogFileName = WCResponseHeadersTest.class.getSimpleName() + ".log";
+        
+        if(!server.isStarted()){
+            server.updateServerConfiguration(savedConfig);
+            server.startServer(consoleLogFileName);
+            Log.info(ME, "setUpBeforeTest", "server started, log file is " + consoleLogFileName);
+        } else if(restoreSavedConfig){
+            try{
+                server.stopServer("CWWKT0042W", "CWWKT0043W", "CWWKT0044W");
+            } catch (Exception e){
+                failure = e;
+            }
+            server.updateServerConfiguration(savedConfig);
+            server.startServer(consoleLogFileName, true);
+            Log.info(getClass(), "setUpBeforeTest", "server restarted, log file is " +consoleLogFileName);
+        }
+        restoreSavedConfig = true;
+        if(failure !=null)
+            throw failure;
+    }
+
+    @BeforeClass
+    public static void setUpOnce() throws Exception {
+
+         //Update the bootstrap to include symbolic default variables for
+        //the <header> attributes. Will be used in testHeaderBootstrapping()
+        writeBootstrapProperty(server, "add.header", "bootstrappedAddHeader:addHeaderValue");
+        writeBootstrapProperty(server, "set.header", "bootstrappedSetHeader:setHeaderValue");
+        writeBootstrapProperty(server, "set.if.missing.header", "bootstrappedSetIfMissingHeader:setIfMissingValue");
+        writeBootstrapProperty(server, "remove.header", "customHeader");
+
+
+        //Create the ResponseHeadersTest.war application
+        ShrinkHelper.defaultDropinApp(server, APP_NAME + ".war", "headers.servlets");
+        savedConfig = server.getServerConfiguration().clone(); 
+        server.setConfigUpdateTimeout(30*1000); 
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        //Stop the server
+        if (server != null ) {
+
+            //Allow the warning messages we may have generated on purpose
+            //
+            // W CWWKT0042W: An empty header name was found when the {0} configuration was parsed. This value is ignored.
+            //
+            // W CWWKT0043W: A duplicate header name was found in the [{0}] header using the {1} configuration. All configurations
+            // for the [{0}] header are ignored. Any header that is defined by the remove, add, set, or setIfMissing configurations
+            //must be unique across all configurations.
+            //
+            // W CWWKT0044W: The [{0}] header, which is marked as a duplicate header name, was found in the {1} configuration.
+            // The [{0}] header is ignored. Any header that is defined by the {1} configuration must contain unique header names.
+            if(server.isStarted())
+                server.stopServer("CWWKT0042W", "CWWKT0043W", "CWWKT0044W");
```

Comment:
Usually we add a `server != null` check here as well.

Reply:
These lines are already wrapped by a `if (server !=null)` block, around line 123 (give or take a few lines since it might be adjusted due to the other suggested changes.